### PR TITLE
Adding support to Matterport3D

### PR DESCRIPTION
**Pull request by hugoycj**
- [x] Add Matterport3D preprocessing
- [x] Add MatterPort3D datasets
- [x] Add semseg-spunet-v1m1-0-base configs for MatterPort3D 
- [x] Update Data Preparation Docs
- [x]  Add preprocessed data and pretrained weights
- [ ]  Fix bugs in testing

### CODE REVIEW

**Comment by Gofinge** [comment]
Thanks for your contribution. I will process your PR this weekend.

**Comment by hugoycj** [comment]
@Gofinge As part of my research, I have remapped Matterport3D's semantic labels to ScanNet 20 semantic labels with the addition of a ceiling label. This was done as a means of assessing the impact of pretraining with Matterport3D on ScanNet performance.  If necessary, I will create new class_id labels specifically for the original Matterport3D semantic labels.

**Comment by Gofinge** [comment]
Thanks. I need some time to review the code (I need to download the dataset first and follow your code). By the way, I am going to publicize one work related to multi-dataset pre-training on arXiv tonight, code is ready, but I won't release it recently. If you find the paper interesting and need the code, feel free to contact me.

**Comment by hugoycj** [comment]
That's great!  I am excited to see how your experiments turn out and how these datasets contribute to the performance.

By the way, I am adding more realistic and synthetic datasets to this repo. This will include realistic datasets such as [RIO](https://github.com/WaldJohannaU/3RScan/tree/master), [Taskonomy](https://github.com/StanfordVL/taskonomy), and some synthetic datasets like [3D-FRONT](https://tianchi.aliyun.com/specials/promotion/alibaba-3d-scene-dataset), and [hypersim](https://github.com/apple/ml-hypersim). Looking forward to seeing how these expanded datasets will contribute to the field of multi-dataset pre-training research

**Comment by Gofinge** [comment]
> That's great! I am excited to see how your experiments turn out and how these datasets contribute to the performance.
> 
> By the way, I am adding more realistic and synthetic datasets to this repo. This will include realistic datasets such as [RIO](https://github.com/WaldJohannaU/3RScan/tree/master), [Taskonomy](https://github.com/StanfordVL/taskonomy), and some synthetic datasets like [3D-FRONT](https://tianchi.aliyun.com/specials/promotion/alibaba-3d-scene-dataset), and [hypersim](https://github.com/apple/ml-hypersim). Looking forward to seeing how these expanded datasets will contribute to the field of multi-dataset pre-training research

Great! Let's build one uniform codebase for 3D perception!

**Comment by hugoycj** [comment]
The preprocessed data have been added to the README, and the training log can be found at this [link](https://cuhko365-my.sharepoint.com/:u:/g/personal/118010378_link_cuhk_edu_cn/EeCdRrrBRh9IqcYwRBlR6fIBmA1aGQr1FoLn9xEPAcBS2Q?e=Hrqtfa).

The evaluation results are summarized in the table below

| Metric | Value |
|--------|-------|
| mIoU | 0.4562 |
| mAcc | 0.5694 |
| allAcc | 0.8024 |

| Class | IoU | Accuracy |
|-------|-----|----------|
| Class_0-wall | 0.7206 | 0.8614 |
| Class_1-floor | 0.9131 | 0.9659 |
| Class_2-cabinet | 0.3793 | 0.6247 |
| Class_3-bed | 0.7327 | 0.9079 |
| Class_4-chair | 0.6730 | 0.8395 |
| Class_5-sofa | 0.6643 | 0.8214 |
| Class_6-table | 0.3614 | 0.5018 |
| Class_7-door | 0.5121 | 0.6742 |
| Class_8-window | 0.4629 | 0.5893 |
| Class_9-bookshelf | 0.3752 | 0.4968 |
| Class_10-picture | 0.3629 | 0.4473 |
| Class_11-counter | 0.2205 | 0.3260 |
| Class_12-desk | 0.0002 | 0.0002 |
| Class_13-curtain | 0.5768 | 0.7084 |
| Class_14-refrigerator | 0.0173 | 0.0178 |
| Class_15-shower curtain | 0.0000 | 0.0000 |
| Class_16-toilet | 0.5070 | 0.6327 |
| Class_17-sink | 0.3662 | 0.5553 |
| Class_18-bathtub | 0.7189 | 0.8641 |
| Class_19-other | 0.1502 | 0.1835 |
| Class_20-ceiling | 0.8648 | 0.9396 |

**Comment by Gofinge** [comment]
I am sorry for the long wait. I just finished downloading the raw Matterport3D dataset and will review your code this weekend! Thanks again for your contribution.

**Comment by Gofinge** [comment]
Hi, @hugoycj , would you mind contacting me with my email (xiaoyang.wu.cs@gmail)? As we recently introduced the Black formatter, we need to format, rebase, and force-push your code before merging. Can we open a Zoom meeting and I can guide you on how to smoothly achieve that?

**Review by Gofinge — APPROVED** [review]

**Comment by Gofinge** [comment]
@hugoycj Sorry for the super super long wait. I finally got some time to handle this PR as I am working on scaling up self-supervised pre-training with more data.

I like your code, it's helpful! Thanks for contributing to Pointcept. I made a few modifications with this commit: https://github.com/Pointcept/Pointcept/pull/69/commits/5258aa668990b18f364f31b515ef9a18924e362f
